app/controllers/two_factor_authentication/sms_opt_in_controller.rb

```diff
@@ -53,14 +54,13 @@ def mfa_context
       @mfa_context ||= MfaContext.new(current_user)
     end
 
-    def load_phone_configuration
-      if user_session.present? && (unconfirmed_phone = user_session[:unconfirmed_phone]).present?
-        @phone_configuration = PhoneConfiguration.new(phone: unconfirmed_phone)
-      elsif user_session.present? && (phone_id = user_session[:phone_id]).present?
-        @phone_configuration = mfa_context.phone_configuration(phone_id)
-      else
-        render_not_found
-      end
+    def load_phone
+      @phone_number_opt_out = PhoneNumberOptOut.find(params[:opt_out_id])
```

Comment:
So... this is a numeric ID in the URL, which is not great... it allows enumeration, so an adventurous person could try to opt back in random phone numbers, which is bad because then it enables spam slightly more

Got a few ideas:
- Separate UUID column?
- Just use the fingerprint column?

Any preferences?

Reply:
went with uuid in ed8facfd56485a66b2efe11dd2d976e3b5a6b25a